In [1]:
# Standard Data Science Helpers
import numpy as np
import pandas as pd
import scipy

import plotly.graph_objs as go
import plotly.graph_objects as go
import plotly.express as px

from ipywidgets import widgets

import chart_studio.plotly as py

from plotly.offline import iplot, init_notebook_mode
init_notebook_mode(connected=True)

import cufflinks as cf
cf.go_offline(connected=True)
cf.set_config_file(colorscale='plotly', world_readable=True)

# Extra options
pd.options.display.max_rows = 30
pd.options.display.max_columns = 25

# Show all code cells outputs
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'

import os
from IPython.display import Image, display, HTML

import ipywidgets as widgets
from ipywidgets import interact, interact_manual

In [2]:
nt_df = pd.read_csv('nt_crime_statistics_may_2022.csv')

# might use Southern Australia data for comparison
sa_df = pd.read_csv('2013-14-data_sa_crime.csv') 

In [86]:
nt_df.keys()

Index(['As At', 'Year', 'Month number', 'Offence category', 'Offence type',
       'Alcohol involvement', 'DV involvement', 'Reporting region',
       'Statistical Area 2', 'Number of offences'],
      dtype='object')

In [87]:
nt_df.columns = [colname.strip() for colname in nt_df.columns]
nt_df.drop('As At', axis = 1)

Year  Month number                                   Offence category  \
0      2008             1  Abduction - harassment and other offences agai...   
1      2008             1  Abduction - harassment and other offences agai...   
2      2008             1  Abduction - harassment and other offences agai...   
3      2008             1  Abduction - harassment and other offences agai...   
4      2008             1  Abduction - harassment and other offences agai...   
...     ...           ...                                                ...   
44989  2022             5         Theft and related offences (other than MV)   
44990  2022             5         Theft and related offences (other than MV)   
44991  2022             5         Theft and related offences (other than MV)   
44992  2022             5         Theft and related offences (other than MV)   
44993  2022             5         Theft and related offences (other than MV)   

                                     Offence type Alcohol involvement  \
0      Deprivation of liberty /false imprisonment                   -   
1      Deprivation of liberty /false imprisonment                   -   
2      Deprivation of liberty /false imprisonment                   -   
3                 Harassment and private nuisance                   -   
4                 Harassment and private nuisance                   -   
...                                           ...                 ...   
44989  Theft and related offences (other than MV)                   -   
44990  Theft and related offences (other than MV)                   -   
44991  Theft and related offences (other than MV)                   -   
44992  Theft and related offences (other than MV)                   -   
44993  Theft and related offences (other than MV)                   -   

      DV involvement Reporting region    Statistical Area 2  \
0                  -    Alice Springs                     -   
1                  -       NT Balance             Alligator   
2                  -    Tennant Creek                     -   
3                  -       NT Balance             Alligator   
4                  -       NT Balance                  Daly   
...              ...              ...                   ...   
44989              -       NT Balance               Weddell   
44990              -       NT Balance           West Arnhem   
44991              -       NT Balance  Yuendumu - Anmatjere   
44992              -       Palmerston                     -   
44993              -    Tennant Creek                     -   

       Number of offences  
0                       1  
1                       1  
2                       1  
3                       2  
4                       1  
...                   ...  
44989                   5  
44990                   8  
44991                   8  
44992                  70  
44993                   7  

[44994 rows x 9 columns]

In [88]:
region_year_offences_df = pd.DataFrame(nt_df.groupby(['Reporting region', 'Year'])['Number of offences'].sum().reset_index())
region_year_offences_df

Reporting region  Year  Number of offences
0      Alice Springs  2008                5312
1      Alice Springs  2009                5735
2      Alice Springs  2010                6989
3      Alice Springs  2011                6235
4      Alice Springs  2012                7418
..               ...   ...                 ...
100    Tennant Creek  2018                1107
101    Tennant Creek  2019                1344
102    Tennant Creek  2020                1277
103    Tennant Creek  2021                1451
104    Tennant Creek  2022                 470

[105 rows x 3 columns]

In [91]:
region_month_year_df = pd.DataFrame(nt_df.groupby(['Reporting region', 'Year', 'Month number'])['Number of offences'].sum().reset_index())
region_month_year_df

Reporting region  Year  Month number  Number of offences
0       Alice Springs  2008             1                 420
1       Alice Springs  2008             2                 358
2       Alice Springs  2008             3                 508
3       Alice Springs  2008             4                 487
4       Alice Springs  2008             5                 406
...               ...   ...           ...                 ...
1206    Tennant Creek  2022             1                  63
1207    Tennant Creek  2022             2                  79
1208    Tennant Creek  2022             3                 126
1209    Tennant Creek  2022             4                 119
1210    Tennant Creek  2022             5                  83

[1211 rows x 4 columns]

In [96]:
region_month_year_offence_category = pd.DataFrame(nt_df.groupby(['Reporting region', 'Year', 'Month number', 'Offence category', 'Offence type'])['Number of offences'].sum().reset_index())
region_month_year_offence_category

Reporting region  Year  Month number  \
0        Alice Springs  2008             1   
1        Alice Springs  2008             1   
2        Alice Springs  2008             1   
3        Alice Springs  2008             1   
4        Alice Springs  2008             1   
...                ...   ...           ...   
14752    Tennant Creek  2022             5   
14753    Tennant Creek  2022             5   
14754    Tennant Creek  2022             5   
14755    Tennant Creek  2022             5   
14756    Tennant Creek  2022             5   

                                        Offence category  \
0      Abduction - harassment and other offences agai...   
1      Abduction - harassment and other offences agai...   
2                          Acts intended to cause injury   
3                                   Commercial break-ins   
4                                   Commercial break-ins   
...                                                  ...   
14752                   Motor theft and related offences   
14753                   Motor theft and related offences   
14754  Other Dangerous or negligent acts endangering ...   
14755                           Property Damage Offences   
14756         Theft and related offences (other than MV)   

                                            Offence type  Number of offences  
0             Deprivation of liberty /false imprisonment                   1  
1                                  Threatening behaviour                   4  
2                                                Assault                  95  
3                            Commercial break-ins actual                  31  
4                         Commercial break-ins attempted                   1  
...                                                  ...                 ...  
14752                     Illegal use of a motor vehicle                   1  
14753           Theft of motor vehicle parts or contents                   2  
14754  Other dangerous or negligent acts endangering ...                   2  
14755                           Property damage offences                  35  
14756         Theft and related offences (other than MV)                   7  

[14757 rows x 6 columns]

In [98]:
offence_category_type = pd.DataFrame(nt_df.groupby(['Offence category','Offence type'])['Number of offences'].sum()).reset_index()
offence_category_type

Offence category  \
0   Abduction - harassment and other offences agai...   
1   Abduction - harassment and other offences agai...   
2   Abduction - harassment and other offences agai...   
3   Abduction - harassment and other offences agai...   
4                       Acts intended to cause injury   
5                       Acts intended to cause injury   
6                                Commercial break-ins   
7                                Commercial break-ins   
8                       Homicide and related Offences   
9                       Homicide and related Offences   
10                      Homicide and related Offences   
11                      Homicide and related Offences   
12                                    House break-ins   
13                                    House break-ins   
14                   Motor theft and related offences   
15                   Motor theft and related offences   
16  Other Dangerous or negligent acts endangering ...   
17                           Property Damage Offences   
18           Robbery - extortion and related offences   
19           Robbery - extortion and related offences   
20                Sexual Assault and related offences   
21                Sexual Assault and related offences   
22         Theft and related offences (other than MV)   

                                         Offence type  Number of offences  
0                            Abduction and kidnapping                  15  
1          Deprivation of liberty /false imprisonment                 610  
2                     Harassment and private nuisance                1088  
3                               Threatening behaviour                3314  
4                                             Assault              102368  
5                 Other acts intended to cause injury                 371  
6                         Commercial break-ins actual               26171  
7                      Commercial break-ins attempted                3429  
8                                    Attempted Murder                  45  
9                               Driving causing death                  37  
10                                       Manslaughter                  45  
11                                             Murder                 121  
12                             House break-ins actual               25953  
13                          House break-ins attempted                3543  
14                     Illegal use of a motor vehicle               13289  
15           Theft of motor vehicle parts or contents               16197  
16  Other dangerous or negligent acts endangering ...                1787  
17                           Property damage offences              108292  
18                            Blackmail and extortion                  28  
19                                            Robbery                1781  
20                     Non-assaultive sexual offences                 796  
21                                     Sexual assault                5452  
22         Theft and related offences (other than MV)              105809

In [105]:
offence_category = pd.DataFrame(nt_df.groupby('Offence category')['Number of offences'].sum()).reset_index()
offence_category

Offence category  Number of offences
0   Abduction - harassment and other offences agai...                5027
1                       Acts intended to cause injury              102739
2                                Commercial break-ins               29600
3                       Homicide and related Offences                 248
4                                     House break-ins               29496
5                    Motor theft and related offences               29486
6   Other Dangerous or negligent acts endangering ...                1787
7                            Property Damage Offences              108292
8            Robbery - extortion and related offences                1809
9                 Sexual Assault and related offences                6248
10         Theft and related offences (other than MV)              105809

In [4]:
#  Get longitude and latitude and concat / join 

In [5]:
nt_df[nt_df['Year'] == 2013].describe()

Year  Month number  Number of offences
count  3103.0   3103.000000         3103.000000
mean   2013.0      6.422172            8.638737
std       0.0      3.472883           23.206805
min    2013.0      1.000000            1.000000
25%    2013.0      3.000000            1.000000
50%    2013.0      6.000000            2.000000
75%    2013.0      9.000000            6.000000
max    2013.0     12.000000          304.000000

[Current - Northern Territory Crime Statistics May 2022](https://data.nt.gov.au/dataset/current-northern-territory-crime-statistics-may-2022)

[South Australian Government Data Directory](https://data.sa.gov.au/data/dataset/crime-statistics)

Offences in the Northern Territory can show strong seasonal patterns, with predictable peaks and troughs at certain times of the year.  Therefore, comparisons should be made of the same months in different years, rather than consecutive months.  For trends, a 12-month rolling average or rolling sum is recommended in order to dampen the month-to-month variability and account for the seasonal cycles.

Last update was July 1st 2022.

In 2017, the governments of the UK, Germany and Canada issued travel advisories to their citizens travelling in Australia, singling out Alice Springs as a place where extra precautions should be taken: "Take care in the town centre of Alice Springs at night. There have been a number of incidents of harassment, robberies and attacks (including sexual assault) on foreign tourists." (UK).

"Foreign travellers warned about visiting Alice Springs". The New Daily. 15 March 2017. Retrieved 9 October 2019.

Webb, Carolyn (15 March 2017). "Foreign tourists warned on Alice Springs safety risks". The Sydney Morning Herald. Retrieved 9 October 2019.

"The Northern Territory is the Murder Capital of Australia." The Guardian. 18 June 2013.   https://www.theguardian.com/news/datablog/2013/jun/18/australia-new-crime-statistics. Retrieved 15 August 2021 

![Murder Capital of Australia](murder_capital.png)

Australia's Northern Territory has acquired quite a reputation due to *its consistently higher per capita rate for homicide and manslaughter than other Australian states and territories, as well as a high level of assaults.* ([Wikipedia](https://en.wikipedia.org/wiki/Crime_in_the_Northern_Territory)) Most of these were related to domestic violence and alcohol intake. 

The headline presented above came from an article which was written for the [Guardian](https://www.theguardian.com/news/datablog/2013/jun/18/australia-new-crime-statistics) by Nick Evershed on the 18th of June in 2013. He wrote that the Northern Territory's murder and assault rates are the highest in the country. 

The aim of this exploratory data analysis is to look into the crime trends in Australia's Northern territory using the Pandas and Plotly. The dataset used was the most recently updated [Current - Northern Territory Crime Statistics May 2022](https://data.nt.gov.au/dataset/current-northern-territory-crime-statistics-may-2022) from the Northern Territory Government website.

<h2 style = 'color: blue'> Reported Offences in the Northern Territory </h2>

In [110]:
px.pie(offence_category, 
       values = 'Number of offences', 
       names = 'Offence category', 
       title = 'Breakdown of Reported Offences in the Northern Territory From 2008 - 2022')

> ### The top 3 reported offences in the Northern Territory from 2008 to 2022 are: 
>
> - Property Damage Offences: 25.8%
> - Theft and Related Offences (other than Motor Vehicle Theft): 25.2%
> - Acts intended to cause injury: 24.4%
>

Homicide and related offences only make up 0.059% of the reported offences. With a total of 44,994 reported offences, this means that 248 incidences of homicide occurred in a span of 13.5 years, which can indicate that an average of 18 homicides are reported each year. 

In [144]:
@interact
def show_offence_type(Category = list(region_month_year_offence_category['Offence category'].unique()),
                      By_Year = False,
                      Year = list(region_month_year_offence_category['Year'].unique()),
                     By_Month = False,
                      Month_Number = list(region_month_year_offence_category['Month number'].unique())):
    
    category_filter = region_month_year_offence_category['Offence category'] == Category
    year_filter = region_month_year_offence_category['Year'] == Year
    month_filter = region_month_year_offence_category['Month number'] == Month_Number
    
    
    if By_Month & By_Year:
        region_month_year_offence_category[category_filter & year_filter & month_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence type', title=f'Types of Reported {Category} in Month {Month_Number} of {Year}')
    elif By_Year and not By_Month:
        region_month_year_offence_category[category_filter & year_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence type', title=f'Types of Reported {Category} in {Year}')    
    elif By_Month and not By_Year:
        region_month_year_offence_category[category_filter & month_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence type', title=f'Types of Reported {Category} in Month {Month_Number} from 2008 to 2022')
    else:
        region_month_year_offence_category[category_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence type', title=f'Types of Reported {Category} from 2008 to 2022')


interactive(children=(Dropdown(description='Category', options=('Abduction - harassment and other offences aga…

In [6]:
print(nt_df['Number of offences'].describe())

count    44994.000000
mean         9.346602
std         26.984379
min          1.000000
25%          1.000000
50%          3.000000
75%          6.000000
max        458.000000
Name: Number of offences, dtype: float64


In [95]:
px.line(region_year_offences_df, x = 'Year', y = 'Number of offences', color = 'Reporting region', markers = True)

<h2 style = 'color: blue'> Yearly Trend </h2>

According to the source of the data, [Current - Northern Territory Crime Statistics May 2022](https://data.nt.gov.au/dataset/current-northern-territory-crime-statistics-may-2022), the *offences in the Northern Territory can show **strong seasonal patterns, with predictable peaks and troughs at certain times of the year.**  Therefore, comparisons should be made of the same months in different years, rather than consecutive months.  For trends, a 12-month rolling average or rolling sum is recommended in order to dampen the month-to-month variability and account for the seasonal cycles.*

In [50]:
@interact
def show_yearly_crime_count(Year=(2008, 2022, 1)):
    
    monthly_sum = nt_df[nt_df['Year']==Year].groupby('Month number')['Number of offences'].sum()
    yearly_mean = nt_df[nt_df['Year']==Year].groupby('Month number')['Number of offences'].sum().mean()
    monthly_variance = monthly_sum - yearly_mean
    maximum_sum = nt_df[nt_df['Year']==Year].groupby('Month number')['Number of offences'].sum().max()
    minimum_sum = nt_df[nt_df['Year']==Year].groupby('Month number')['Number of offences'].sum().min()
    years = nt_df['Year'].unique
    
    yearly_count = pd.DataFrame(monthly_sum)
    yearly_count['Mean'] = yearly_mean
    yearly_count['Variance'] = monthly_variance 
    yearly_count['Percent Variance'] = monthly_variance / yearly_mean * 100
    
    yearly_max_min = pd.DataFrame([maximum_sum, minimum_sum])
    print(yearly_max_min, yearly_count)
    
    if len(yearly_count.index) > 5:
        yearly_count.index = ['Jan', 'Feb', 'March', 'April', 'May', 'June', 'July', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec']
    else:
        yearly_count.index = ['Jan', 'Feb', 'March', 'April', 'May']
                 

    yearly_count.iplot(kind='bar',
                       barmode='group', 
                       title=f'Number of Offences, Mean & Variance For Each Month in {Year}'
                      )
    

interactive(children=(IntSlider(value=2015, description='Year', max=2022, min=2008), Output()), _dom_classes=(…

In [11]:
@interact
def show_monthly_crime_count(Month=[('January', 1), ('February', 2), ('March', 3), ('April', 4), ('May', 5), ('June', 6), ('July', 7), ('August', 8), ('September', 9), ('October', 10), ('November', 11), ('December', 12)]):
    

    monthly_sum_per_year = nt_df[nt_df['Month number'] == Month].groupby('Year')['Number of offences'].sum()
    mean_by_month = nt_df[nt_df['Month number'] == Month].groupby('Year')['Number of offences'].sum().mean()
    variance_by_month = monthly_sum_per_year - mean_by_month
    
    monthly_count = pd.DataFrame(monthly_sum_per_year)
    monthly_count['Mean'] = mean_by_month
    monthly_count['Variance'] = variance_by_month 
    monthly_count['Percent Variance'] = variance_by_month / mean_by_month * 100
    
    print(f'Maximum variance is in ')

    monthly_count.iplot(kind='bar',
                       barmode='group', 
                       title=f'Number of Offences, Mean & Variance for Each Month Grouped By Year'
                      )



interactive(children=(Dropdown(description='Month', options=(('January', 1), ('February', 2), ('March', 3), ('…

In [12]:
years = nt_df['Year'].unique()
months = nt_df['Month number'].unique()
offence_sum = []

for month in months: 
    offence_sum.append(nt_df[nt_df['Month number'] == month].groupby('Year')['Number of offences'].sum())

Offence_sum_df = pd.DataFrame(offence_sum)
Offence_sum_df = Offence_sum_df.set_index(months)
Offence_sum_df[2022] = Offence_sum_df[2022].fillna(0)

In [13]:
Offence_sum_df = Offence_sum_df.transpose()

In [14]:
Offence_sum_df = Offence_sum_df.stack().reset_index()

In [15]:
Offence_sum_df = Offence_sum_df.rename(columns={'level_1': 'Month', 0: 'Number of Offences'})

In [16]:
Offence_sum_df 

Year  Month  Number of Offences
0    2008      1              2420.0
1    2008      2              2417.0
2    2008      3              2630.0
3    2008      4              2583.0
4    2008      5              2262.0
..    ...    ...                 ...
175  2022      8                 0.0
176  2022      9                 0.0
177  2022     10                 0.0
178  2022     11                 0.0
179  2022     12                 0.0

[180 rows x 3 columns]

In [17]:
fig1 = px.violin(Offence_sum_df[Offence_sum_df['Year'] != 2022], 
                x="Month", 
                y="Number of Offences", 
                color='Month', 
                box=True, 
                points='all',
                title = 'A Comparison of Reported Offences By Month From 2008 to 2021')

fig1.show()


In [23]:
Offence_sum_df_no_2020 = Offence_sum_df[(Offence_sum_df['Year'] != 2020) & (Offence_sum_df['Year'] != 2022)] 
fig1 = px.violin(Offence_sum_df_no_2020, 
                x="Month", 
                y="Number of Offences", 
                color='Month', 
                box=True, 
                points='all',
                title = 'A Comparison of Reported Offences By Month From 2008 to 2019, 2021 ')

fig1.show()

In [222]:
year_month_df = pd.DataFrame(nt_df.groupby(['Year', 'Month number']).agg({'Number of offences': ['mean', 'min', 'max', 'sum']}))

In [171]:
year_month_df.reset_index()

Year Month number Number of offences               
                                     mean min  max   sum
0    2008            1           9.718876   1  340  2420
1    2008            2          10.155462   1  333  2417
2    2008            3          11.385281   1  392  2630
3    2008            4          10.373494   1  360  2583
4    2008            5           9.425000   1  316  2262
..    ...          ...                ...  ..  ...   ...
168  2022            1          11.501706   1  311  3370
169  2022            2          10.142857   1  248  2698
170  2022            3          11.464164   1  296  3359
171  2022            4          11.698582   1  320  3299
172  2022            5           9.884058   1  258  2728

[173 rows x 6 columns]

NameError: name 'year_month_df' is not defined

In [ ]:
year_df = pd.DataFrame(nt_df.groupby('Year').agg({'Number of offences': ['mean', 'min', 'max', 'sum']}))

In [ ]:
year_df

In [ ]:
year_df.keys()

In [ ]:
year_df.index

In [ ]:
year_df[year_df.index == 2008]

<p style='color: red'> The number of offences is the number of offences recorded per region per statistical area. The highest number of offences ever recorded is 458 property damage offences in Darwin in January 2010. </p>

* group together reporting region and statistical area.
* statistical area with maximum offences
* look into the type of offence prevalent in statistical area.
* look into the maximum offences per region.

In [ ]:
nt_df[nt_df['Number of offences'] == 458]

In [ ]:
nt_df[(nt_df['Year'] == 2010) & (nt_df['Month number'] == 1) & 
      (nt_df['Offence category'] == 'Property Damage Offences')]

In [ ]:
nt_df.shape

In [ ]:
nt_df.info()

In [ ]:
nt_df = nt_df.drop('As At', axis = 1)

In [ ]:
nt_df.keys()

<h1 style="color:blue">Offence Category Counts</h1>

<span style="color:blue"> Although it is a relief that homicide and related offences have the least amount of occurences, it is quite alarming that acts intended to cause injury has the highest count among the crimes reported.</span> 

In [ ]:
nt_df.groupby('Offence category')['Number of offences'].sum()

In [ ]:
@interact
def show_offences_per_reporting_region(Region = list(nt_df['Reporting region'].unique())):
    display(HTML(f'<h2 style ="color:purple"> Offences in {Region} from 2008 - 2022 <h2>'))
    nt_df[nt_df['Reporting region'] == Region].groupby('Offence category')['Number of offences'].sum().iplot(kind = 'barh',
                                                                                                             xTitle = 'Month', 
                                                                                                             yTitle = 'Number of Offences')

In [29]:
@interact
def show_offences_per_reporting_region(Region = list(nt_df['Reporting region'].unique())):
    display(HTML(f'<h2 style ="color:purple"> Number of Offences in {Region} from 2008 - 2022 <h2>'))
    nt_df[nt_df['Reporting region'] == Region].groupby('Offence category')['Number of offences'].sum().iplot(kind = 'barh',
                                                                                                             xTitle = 'Month', 
                                                                                                             yTitle = 'Number of Offences')

interactive(children=(Dropdown(description='Region', options=('Alice Springs', 'NT Balance', 'Tennant Creek', …

In [ ]:
nt_df.groupby('Offence category')['Number of offences'].sum().plot(kind = 'barh')

In [ ]:
nt_df[nt_df['Year'] == 2013].groupby('Offence category')['Number of offences'].sum().plot(kind = 'barh')

In [ ]:
nt_df.groupby('Offence category')['Number of offences'].sum()

<h1 style='color:blue'> Which region has the highest number of offences? </h1> 
<p style='color:blue'> The pie chart below shows that majority of the cases reported occurred in NT Balance.</p>

In [ ]:
nt_df[nt_df['Offence category'] == 'Homicide and related Offences'].groupby('Reporting region ')['Number of offences'].sum()

In [ ]:
nt_df[dv_df['Offence category'] == 'Homicide and related Offences'].groupby(
    'Reporting region ')['Number of offences'].sum().plot(kind = "pie", 
                                                          figsize=(20,10), 
                                                          autopct='%1.1f%%')

<p style='color: red'> There were 28,878 cases out of the total cases that were left blank. It can be assumed here that the alcohol involvement was unknown. Replace all - with Alcohol involvement unknown</p>

In [ ]:
nt_df.groupby('Alcohol involvement')['Number of offences'].sum()

<p style='color: red'> There were 28,878 cases out of the total cases that were left blank. It can be assumed here that the domestic violence involvement was unknown. Replace all - with Domestic violence involvement unknown</p>

In [ ]:
nt_df.groupby('DV involvement')['Number of offences'].sum()

In [ ]:
# look into crime by date
#  look into crime by territory
#  crime by season

In [ ]:
nt_df.groupby('Year')['Number of offences'].sum()

In [ ]:
nt_df[nt_df['Month number'] == 1].groupby(['Year'])['Number of offences'].sum().plot(kind = 'bar')

In [ ]:
dv_df[dv_df['Offence category'] == 'Abduction - harassment and other offences against the person']['Number of offences'].sum()

<p style='color: red'> Metadata states that the number of offences fluctuate depending on season but there is not much deviation in the number of offences reported for each month from 2008 - 2022.</p>
* Create a widget that shows bar graph per year. 
* Create bar graph with month in different colors showing crime fluctuation by month and year.

In [ ]:
nt_df[nt_df['Year'] == 2022].groupby('Month number')['Number of offences'].sum()

In [ ]:
nt_df[nt_df['Year'] == 2022].groupby('Month number')['Number of offences'].sum().plot(kind = 'bar')

In [ ]:
nt_df[nt_df['Year'] == 2021].groupby('Month number')['Number of offences'].sum().plot(kind = 'bar')

In [ ]:
nt_df[nt_df['Year'] == 2020].groupby('Month number')['Number of offences'].sum().plot(kind = 'bar')

In [ ]:
nt_df[nt_df['Year'] == 2019].groupby('Month number')['Number of offences'].sum().plot(kind = 'bar')

In [ ]:
nt_df[nt_df['Year'] == 2018].groupby('Month number')['Number of offences'].sum()

In [ ]:
nt_df[nt_df['Year'] == 2017].groupby('Month number')['Number of offences'].sum().plot(kind = 'bar')

In [ ]:
nt_df[nt_df['Year'] == 2016].groupby('Month number')['Number of offences'].sum().plot(kind = 'bar')

In [ ]:
nt_df[nt_df['Year'] == 2015].groupby('Month number')['Number of offences'].sum().plot(kind = 'bar')

In [ ]:
nt_df[nt_df['Year'] == 2014].groupby('Month number')['Number of offences'].sum().plot(kind = 'bar')

In [ ]:
nt_df['Year'].unique()

In [ ]:
nt_df[(nt_df['Year'] == 2013) & (nt_df['Reporting region '] == 'Alice Springs')].groupby('Offence category')['Number of offences'].sum()

In [ ]:
nt_df[nt_df['Year'] == 2022].groupby('Month number')['Number of offences'].sum()

In [ ]:
# data.columns = [colname.strip() for colname in data.columns]